process_manager/views/partials.py

```diff
@@ -63,31 +62,39 @@ def process_table(request: HttpRequest) -> HttpResponse:
 
     status_enum_lookup = dict(item[::-1] for item in ProcessInstance.StatusCode.items())
 
-    table_data = []
-    process_instances = session_info.data.values
-    for process_instance in process_instances:
-        metadata = process_instance.process_description.metadata
-        uuid = process_instance.uuid.uuid
-        table_data.append(
-            {
-                "uuid": uuid,
-                "name": metadata.name,
-                "user": metadata.user,
-                "session": metadata.session,
-                "status_code": status_enum_lookup[process_instance.status_code],
-                "exit_code": process_instance.return_code,
-            }
+    # Build the table data
+    table_data = [
+        {
+            "uuid": process_instance.uuid.uuid,
+            "name": process_instance.process_description.metadata.name,
+            "user": process_instance.process_description.metadata.user,
+            "session": process_instance.process_description.metadata.session,
+            "status_code": status_enum_lookup[process_instance.status_code],
+            "exit_code": process_instance.return_code,
+        }
+        for process_instance in session_info.data.values
+    ]
+    # Get the values from the GET request
+    search_dropdown = request.GET.get("search-drp", "")
+    search_input = request.GET.get("search", "")
+
+    # Combine them into filter_query, joining by ':' only if search_input is provided
+    if search_dropdown and not search_input:
+        filter_query = ""
+    else:
+        filter_query = (
+            f"{search_dropdown}:{search_input}" if search_input else search_dropdown
         )
-    # Filter table data based on search parameter
-    table_data = filter_table(request.GET.get("search", ""), table_data)
+    # Apply search filtering
+    table_data = filter_table(filter_query, table_data)
```

Comment:
Now that we have the column directly here, we don't need any of this to create a combined search term. Modify the `filter_table` function so it takes the column name as parameter as well and [remove the code within that function](https://github.com/ImperialCollegeLondon/drunc_ui/blob/fcd25490b4834c1c851aac3f9de7078ddbb42713/process_manager/views/partials.py#L38) that precisely split the search query into column and search term.